domain/schema/model/sql/0019-application.sql

```diff
@@ -279,3 +279,32 @@ SELECT
     a.application_uuid
 FROM application_endpoint AS a
 JOIN charm_relation AS c ON a.charm_relation_uuid = c.uuid;
+
+-- v_principal_subordinate returns the uuids of an subordinate application
+-- and its principal application. A principal application may multiple
+-- subordinates as a subordinate may be related to multiple principals.
+-- Find the specific principal application for a unit in the unit_principal
+-- table.
+CREATE VIEW v_principal_subordinate AS
```

Comment:
Let me have a go at rewriting this. I think I can make it more efficient.

Reply:
@manadart Due to removal of the method using it,  I've deleted this view.